pkg/repository/udmrepo/repo_options.go

```diff
@@ -200,7 +200,12 @@ func GetRepoDomain() string {
 
 func getRepoConfigFile(workPath string, repoID string) string {
 	if workPath == "" {
-		workPath = filepath.Join(os.Getenv("HOME"), "udmrepo")
+		home := os.Getenv("HOME")
```

Comment:
For the random uid case, is it the env var not set or it's set but the directory does not exist?

Reply:
The env var is not set. See the reported error in #8067:

`mkdir /udmrepo: permission denied`

Reply:
ok